tensor2tensor/data_generators/g2p.py

```diff
@@ -0,0 +1,136 @@
+# coding=utf-8
+# Copyright 2017 The Tensor2Tensor Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data generators for g2p (grapheme to phoneme) task."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import os
+import sys
+import tarfile
+
+# Dependency imports
+
+from tensor2tensor.data_generators import generator_utils
+from tensor2tensor.data_generators import problem
+from tensor2tensor.data_generators import text_encoder
+from tensor2tensor.utils import registry
+
+import tensorflow as tf
+
+EOS = text_encoder.EOS
+EOS_ID = text_encoder.EOS_ID
+CMUDICT_URL = "https://sourceforge.net/projects/cmusphinx/files/G2P%20Models/phonetisaurus-cmudict-split.tar.gz/download"
+
+
+def _build_vocab(filename, vocab_path):
+  """Reads a file to build a vocabulary with letters and phonemes.
+
+  Args:
+    filename: file to read list of words from.
+    vocab_path: path where to save the vocabulary.
+  """
+  vocab = set()
+  with tf.gfile.GFile(filename, "r") as f:
+    for line in f:
+        items = line.split()
+        vocab.update(items[0])
+        vocab.update(items[1:])
+  with open(vocab_path, "w") as f:
+    f.write("<pad>\n")
+    f.write("<EOS>\n")
+    f.write("\n".join(sorted(vocab)))
+
+
+def _get_token_encoder(vocab_dir, vocab_name, filename):
+  """Reads from file and returns a `TokenTextEncoder` for the vocabulary."""
+  vocab_path = os.path.join(vocab_dir, vocab_name)
+  if not tf.gfile.Exists(vocab_path):
+    _build_vocab(filename, vocab_path)
+  return text_encoder.TokenTextEncoder(vocab_path)
+
+@registry.register_problem
+class G2p(problem.Text2TextProblem):
```

Comment:
this name seems a bit too short for comfort. Grapheme2Phoneme, or GraphemeToPhoneme